moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -969,6 +963,122 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
     }
   }
 
+  return doTimeParameterizationCalculations(trajectory, max_velocity, max_acceleration);
+}
+
+bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotTrajectory& trajectory,
+                                                        const std::map<std::string, double>& velocity_limits,
+                                                        const std::map<std::string, double>& acceleration_limits) const
+{
+  if (trajectory.empty())
+    return true;
+
+  // Get the default joint limits from the robot model, then overwrite any that are provided as arguments
+  const moveit::core::JointModelGroup* group = trajectory.getGroup();
+  if (!group)
+  {
+    RCLCPP_ERROR(LOGGER, "It looks like the planner did not set the group the plan was computed for");
+    return false;
+  }
+  const unsigned num_joints = group->getVariableCount();
+  const std::vector<std::string>& vars = group->getVariableNames();
+  const moveit::core::RobotModel& rmodel = group->getParentModel();
+
+  Eigen::VectorXd max_velocity(num_joints);
+  Eigen::VectorXd max_acceleration(num_joints);
+  for (size_t j = 0; j < num_joints; ++j)
+  {
+    const moveit::core::VariableBounds& bounds = rmodel.getVariableBounds(vars[j]);
+
+    // VELOCITY LIMIT
+    // Check if a custom limit was specified as an argument
+    bool set_velocity_limit = false;
+    for (const auto& [joint_name, velocity_limit] : velocity_limits)
+    {
+      if (joint_name.compare(vars[j]))
+      {
+        max_velocity[j] = velocity_limit;
+        set_velocity_limit = true;
+        break;
+      }
+    }
```

Comment:
Could this iteration be replaced by a simple lookup like this, or am I missing something?
```
auto it = velocity_limits.find(vars[j]);
if (it != velocity_limits.end()) {
  max_velocity[j] = it->second;
  set_velocity_limit = true;
}
```